includes/class-affiliatewp-leaderboard.php

```diff
@@ -230,26 +239,42 @@ public function affiliate_leaderboard( $atts, $content = null ) {
 	 */
 	public function show_leaderboard( $args = array() ) {
 
-		$defaults = apply_filters( 'affwp_leaderboard_defaults',
-			array(
-				'number'  => isset( $args['number'] ) ? $args['number'] : 10,
-				'orderby' => isset( $args['orderby'] ) ? $args['orderby'] : 'referrals',
+		/**
+		 * Filters the leaderboard shortcode attributes.
+		 *
+		 * @since 1.0
+		 *
+		 * @param array $attributes Shortcode attributes.
+		 *    $atts = [
+		 *      'earnings'  => (string) Toggle earnings visibility.
+		 *      'referrals' => (string) Toggle referrals visibility.
+		 *      'visits'    => (string) Toggle visits visibility.
+		 *      'number'    => (int)    Number of items to display.
+		 *      'orderby'   => (string) DB column to order results.
+		 *    ]
+		 */
+		$args = wp_parse_args(
+			$args,
+			apply_filters(
+				'affwp_leaderboard_defaults',
+				array(
+					'number'  => isset( $args['number'] ) ? $args['number'] : 10,
+					'orderby' => isset( $args['orderby'] ) ? $args['orderby'] : 'referrals',
```

Comment:
```suggestion
					'orderby' => isset( $args['orderby'] && in_array( $args['orderby'], array( 'values', 'this', 'could', 'be' ), true ) ) ? $args['orderby'] : 'referrals',
```

You can also make sure these are a specific value (sanitize) using `is_array()`.